suricata/update/parsers.py

```diff
@@ -239,3 +233,34 @@ def parse_arg():
             setattr(args, arg, getattr(global_args, arg))
 
     return args
+
+
+def show_all_help():
+  update_arg[2][1]['help'] = "Output YAML fragment for rule inclusion"
+  update_arg[5][1]['help'] = "Generate a sid-msg.map file"
+  update_arg[6][1]['help'] = "Generate a v2 sid-msg.map file"
+  update_arg[11][1]['help'] = "Filenames to ignore (can be specified multiple times; default: *deleted.rules)"
+  update_arg[12][1]['help'] = "Disables the ignore option."
+  update_arg[13][1]['help'] = "Filename of rule thresholding configuration"
+  update_arg[14][1]['help'] = "Output of processed threshold configuration"
+  update_arg[15][1]['help'] = "Dump sample config files to current directory"
+  update_arg[21][1]['help'] = "Do not merge the rules into a single file"
```

Comment:
I think this will be fragile against changes to the arguments - the indexes might get out of sync. I had something more in mind like:
```
show_advanced = False
```
```
     (("--sid-msg-map",),
-     {'metavar': '<filename>', 'help': argparse.SUPPRESS}),
+     {'metavar': '<filename>', 'help': "Generate a sid-msg.map file" if show_advanced else argparse.SUPPRESS}),
     (("--sid-msg-map-2",),
```

I guess there is the issue of setting `show_advanced` to True. This will probably have to be done before parsing any args with a quick scan through argv..
```
if "-s" in sys.argv or "--show-all-options" in sys.argv:
    show_advanced = True
```

Reply:
That's a great way to solve the issue!